visual-dotnet/SauceLabs.Visual/VisualClient.cs

```diff
@@ -92,6 +120,15 @@ public async Task<string> VisualCheck(VisualBuild build, string name, VisualChec
             return result.Result.Id;
         }
 
+        public void Dispose()
+        {
+            _api.Dispose();
+            if (!_externalBuild)
+            {
+                FinishBuild(Build).Wait();
```

Comment:
uh never use Wait. This might cause dead locks: https://learn.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#async-all-the-way